src/components/CopyToClipboard/CopyToClipboard.tsx

```diff
@@ -7,75 +7,46 @@ export enum CopyToClipboardStatus {
     Error = 'error',
 }
 
-interface CopyToClipboardGeneralProps {
+export interface CopyToClipboardProps {
     children: (status: CopyToClipboardStatus) => React.ReactElement;
     text: string;
+    timeout?: number;
     onCopy?: (text: string, result: boolean) => void;
 }
 
-interface CopyToClipboardDefaultProps {
-    timeout: number;
-}
-
-interface CopyToClipboardInnerProps
-    extends CopyToClipboardGeneralProps,
-        CopyToClipboardDefaultProps {}
-
-export interface CopyToClipboardProps
-    extends CopyToClipboardGeneralProps,
-        Partial<CopyToClipboardDefaultProps> {}
-
-interface CopyToClipboardState {
-    status: CopyToClipboardStatus;
-}
-
-export class CopyToClipboard extends React.Component<
-    CopyToClipboardInnerProps,
-    CopyToClipboardState
-> {
-    static INITIAL_STATUS = CopyToClipboardStatus.Pending;
-
-    state: CopyToClipboardState = {
-        status: CopyToClipboard.INITIAL_STATUS,
-    };
-
-    private timerId?: number;
-
-    render() {
-        const {children, text} = this.props;
-        const {status} = this.state;
-        const content = children(status);
-
-        if (!React.isValidElement(content)) {
-            throw new Error('Content must be a valid react element');
+export function CopyToClipboard({children, text, timeout = 1000, onCopy}: CopyToClipboardProps) {
+    const [status, setStatus] = React.useState(CopyToClipboard.INITIAL_STATUS);
+    const content = React.useMemo(() => children(status), [children, status]);
+    const handleCopy = React.useCallback<Required<ReactCopyToClipboard.Props>['onCopy']>(
+        (copyText, result) => {
+            setStatus(result ? CopyToClipboardStatus.Success : CopyToClipboardStatus.Error);
+            onCopy?.(copyText, result);
+        },
+        [onCopy],
+    );
```

Comment:
Let's not change the current behaviour. 

```suggestion
    const timer = React.useRef<number>();
    const handleCopy = React.useCallback<Required<ReactCopyToClipboard.Props>['onCopy']>(
        (copyText, result) => {
            setStatus(result ? CopyToClipboardStatus.Success : CopyToClipboardStatus.Error);
            window.clearTimeout(timer.current);
            timer.current = window.setTimeout(() => {
                setStatus(CopyToClipboard.INITIAL_STATUS);
            }, timeout);
            onCopy?.(copyText, result);
        },
        [onCopy, timeout],
    );
    
    React.useEffect(() => () => {
        clearTimeout(timer.current);
    }, []);
```

Reply:
Why do it in useEffect when it can be done explicitly in the event handler?

Reply:
You are using effect anyway to cleanup timer. Keep logic for reseting status in single place looks tidy.